### DISCUSSION TITLE
Index page with optional route parameters without breaking other subroutes, with redirect to default route parameter

### DISCUSSION THREAD

**Discussion by mklueh** [nuxt/nuxt#16787] [Questions]
I'm running a blog which has a list of all articles on the root page / .

There is also pagination, so pages will be adressed with /1, /2, /3 and so on and the actual blog posts are available under /blog/this-is-one-post.

I want to achieve visiting "/" will show page one of results (/1), but I have no sophisticated solution found yet.

This is my structure, consisting of [page].vue for every route with a page number route parameter. It won't be called when I just go to "/"

<img width="172" alt="image" src="https://user-images.githubusercontent.com/1553757/190269285-cd24f420-4bbe-408f-9df4-2ec4908126ae.png">

For that case I introduced another page index.vue which performs a simple redirect to the first paginated page

```
<script setup>
import {navigateTo} from "#imports";
navigateTo("/1")
</script>
```

This works, but it causes an additional redirect on first page visit.

I've seen there is the option to do this [...page].vue which will allow optional route parameters, but at the cost that now even /blog/... routes will be captured by this view.

Is there any good solution to do what I want to achieve with just one page and no additional redirect?

**Comment by sevillaarvin** [comment]
Hmm, what if you remove `index.vue` and make the page optional with `[[page]].vue`? https://v3.nuxtjs.org/guide/directory-structure/pages#dynamic-routes

Then in your `[[page]].vue` if page is not defined, just set the default to 1?

**Reply by mklueh** [reply]
Thanks for your answer. However, this leads to 404 errors in my case if no parameter is provided
